pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -161,3 +195,28 @@ func (e *Executor) getMentionedAccounts(event model.NotificationEventType) ([]st
 
 	return notification.FindSlackAccounts(event), nil
 }
+
+func (e *Executor) validateApproverNum(approver string, num int) (string, bool) {
+	if num <= 1 {
+		e.LogPersister.Infof("Got approval from \"%s\"", approver)
+		return approver, true
+	}
+	as, ok := e.MetadataStore.Stage(e.Stage.Id).Get(approversKey)
+	if !ok {
+		e.LogPersister.Infof("Got approval from \"%s\"", approver)
+		e.LogPersister.Infof("Waiting for other approvers...")
+		return approver, false
+	}
+	approvers := strings.Split(as, " ,")
+	for _, a := range approvers {
+		if a == approver {
+			return "", false
+		}
+	}
+	e.LogPersister.Infof("Got approval from \"%s\"", approver)
+	if d := num - len(approvers) - 1; d > 0 {
+		e.LogPersister.Infof("Waiting for other approvers...")
+		return as + ", " + approver, false
+	}
```

Comment:
```suggestion
	if c := num - len(approvers); c > 0 {
		e.LogPersister.Infof("Waiting for %d other approvers...", c)
		return as + ", " + approver, false
	}
```

Reply:
Is it intentional that you deleted `-1`?

Reply:
Oh, never mind.
I understand what you want.